### PR TITLE
Draft and lifecycles directories cleanup

### PR DESCRIPTION
**Pull request by nventuro**
Part of https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2060

This PR is best reviewed commit by commit.

It achieves the following:
 - Moves `Pausable` into `utils`, removes `lifecycles`
 - Moves `Strings` and `Counters` from `drafts` into `utils`
 - Moves `SignedSafeMath` into `math`
 - Moves `ERC20Snapshot` into `ERC20`, makes `snapshot` an internal function
 - Removes the `drafts` directory
 - Adds a deprecation notice for `drafts`

https://github.com/OpenZeppelin/docs.openzeppelin.com/pull/160 creates the required redirects.

### CODE REVIEW

File: contracts/token/ERC20/ERC20Snapshot.sol [hunk 1]
```diff
@@ -87,26 +89,27 @@ contract ERC20Snapshot is ERC20 {
...
+        //  requested id, and its value is the one to return.
+        //  c) More snapshots were created after the requested one, and the queried value was later modified. There will be
+        //  no entry for the requested id: the value that corresponds to it is that of the smallest snapshot id that is
+        //  larger than the requested one.
+        //
+        // In summary, we need to find an element in an array, returning the index of the smallest value that is larger if
+        // it is not found, unless said value doesn't exist (e.g. when all values are smaller). Arrays.findUpperBound does
+        // exactly this.
+
         uint256 index = snapshots.ids.findUpperBound(snapshotId);
```
Comment (frangio) [inline contracts/token/ERC20/ERC20Snapshot.sol]: I recall you mentioning that the semantics of `findUpperBound` were very weird and very specific to `ERC20Snapshot`. What do you think about removing the `Arrays` library (that only contains `findUpperBound`) and moving the function to a private one in this contract.

File: contracts/utils/Strings.sol [hunk 2]
```diff
@@ -7,10 +7,11 @@ pragma solidity ^0.6.0;
 library Strings {
     /**
      * @dev Converts a `uint256` to a `string`.
```
Comment (frangio) [inline contracts/utils/Strings.sol]:
```suggestion
     * @dev Converts a `uint256` to its ASCII `string` representation.
```

File: docs/modules/ROOT/pages/drafts.adoc [hunk 3]
```diff
@@ -0,0 +1,19 @@
+= Drafts
+
+All draft contracts were either moved into a different directory or removed from the OpenZeppelin Contracts library on the https://forum.openzeppelin.com/t/openzeppelin-contracts-v3-0-beta-release/2256[v3.0.0 release].
+
+* `ERC20Migrator`: removed.
+* xref:api:token/ERC20.adoc#ERC20Snapshot[`ERC20Snapshot`]: moved to `token/ERC20`.
+* `ERC20Detailed` and `ERC1046`: removed.
+* `TokenVesting`: removed.
```
Comment (frangio) [inline docs/modules/ROOT/pages/drafts.adoc]:
```suggestion
* `TokenVesting`: removed. Pending a replacement that is being discussed in https://github.com/OpenZeppelin/openzeppelin-contracts/issues/1214[`#1214`].
```

File: docs/modules/ROOT/pages/drafts.adoc [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+* xref:api:utils.adoc#Counters[`Counters`]: moved to `utils`.
+* xref:api:utils.adoc#Strings[`Strings`]: moved to `utils`.
+* xref:api:utils.adoc#SignedSafeMath[`SignedSafeMath`]: moved to `utils`.
```
Comment (frangio) [inline docs/modules/ROOT/pages/drafts.adoc]:
```suggestion
* xref:api:utils.adoc#Counters[`Counters`]: moved to xref:api:utils.adoc[`utils`].
* xref:api:utils.adoc#Strings[`Strings`]: moved to xref:api:utils.adoc[`utils`].
* xref:api:utils.adoc#SignedSafeMath[`SignedSafeMath`]: moved to xref:api:utils.adoc[`utils`].
```

File: docs/modules/ROOT/pages/drafts.adoc [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+Removed contracts are still available on the v2.5 release of OpenZeppelin Contracts, which you can install by running:
+
+```console
+$ npm install @openzeppelin/contracts@v2.5
+```
+
+Refer to the https://docs.openzeppelin.com/contracts/2.x/api/utils[v2.x documentation] when working with them.
```
Comment (frangio) [inline docs/modules/ROOT/pages/drafts.adoc]:
```suggestion
Refer to the xref:2.x@contracts:api:utils.adoc[v2.x documentation] when working with them.
```

File: contracts/token/ERC20/ERC20Snapshot.sol (hunk 1 above)
Comment (nventuro) [reply contracts/token/ERC20/ERC20Snapshot.sol]: Yes, this is a good point. I wouldn't mind adding more functions to `Arrays` though, but I cannot think of any useful ones at the moment.

Something that slightly worries me is testing: we'd have to drop our extensive test suite for `findUpperBound` if we were to do this change as proposed.

Comment (frangio) [reply contracts/token/ERC20/ERC20Snapshot.sol]: Yeah that's an interesting point... It's not that I worry about it being the only function in `Array`. I think `Array` could be a useful library. The issue I'm pointing out is that the semantics of `findUpperBound` are maybe not generally useful. We can open an issue to discuss this and tackle it before the release.

Comment (nventuro) [reply contracts/token/ERC20/ERC20Snapshot.sol]: Agreed. I created https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2131

**Pull request merged**